common/loop.c

```diff
@@ -118,8 +118,13 @@ void loop_add_fd(struct loop *loop, int fd, short mask,
 
 	if (loop->fd_length == loop->fd_capacity) {
 		loop->fd_capacity += 10;
-		loop->fds = realloc(loop->fds,
+		struct pollfd *tmp = realloc(loop->fds,
 				sizeof(struct pollfd) * loop->fd_capacity);
+		if (!tmp) {
```

Comment:
Sorry, this dropped off my review queue -- last nit here, rather than assigning to `tmp` first, you can just assign to `loop->fds` and `!loop->fds`.

Reply:
I don’t think so, not at least in the general case. If `realloc` fails, it returns `NULL`. If we write to the `loop->fds` pointer in that case, the memory that was pointed to is lost and there’s a leak.
The patch I’m proposing is idiomatic and lets the possibility of free'ing the memory if realloc fails.

Reply:
Hey, what do you think of it now?